src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.GenerateRootId.netstandard21.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    partial class Activity
+    {
+        private static string GenerateRootId()
+        {
+            // It is important that the part that changes frequently be first, because
+            // some sampling functions don't sample from the high entropy part of their hash function.
+            // This makes sampling based on this produce poor samples.
+            Span<char> result = stackalloc char[1 + 16 + s_uniqSuffix.Length]; // max length needed
```

Comment:
Add a Debug.Assert that provides an upper bound on `s_uniqSuffix.Length` since its being used for stackalloc.

Looks like its 2 + 20 (`MaximumFormatInt64Length = 20;   // 19 + sign (i.e. -9223372036854775808)`)

https://github.com/dotnet/corefx/blob/fd0c739d6bf90ad15020b4f5383d8194dc5ff7d2/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L800

Reply:
Done